src/popover/popover.coffee

```diff
@@ -0,0 +1,108 @@
+###*
+Popover directive displays transcluded content on custom action
+executed on the parent element.
+
+@module shift.components.popover
+
+@param {string} shiftPopover the text of the popover
+@param {string} shiftPopoverTitle (optional) The title of the popover
+@param {string} shiftPopoverTrigger What triggers the popover ? click, hover
+or focus.
+@param {string} shiftPopoverPosition Default to 'bottom', can also be set to
+left, right and top.
+@param {string} shiftPopoverTemplateUrl the template URL for rendering. When
+provided, the text and title attribute are ignored.
+@param {attribute} fixed Use fixed positioning for the the tooltip. To be set
+when the trigger object is also fixed positioned.
+@param {string} shiftPopoverAttachTo a CSS selector where to put the popover
+element. Useful when the popover is appearing in a scrollable area.
+
+@example
+```jade
+input(
+  type = "text"
+  ng-model = "foo"
+  shift-popover = "lorem ipsum"
+  shift-popover-title = "date"
+  shift-popover-trigger = "click|hover|focus"
+  shift-popover-position = "top|bottom|left|right"
+  shift-popover-template-url = "xyz.html"
+  shift-popover-attach-to = ".scrollable.classname"
+  fixed
+)
+```
+###
+
+angular.module 'shift.components.popover', []
+  .directive 'shiftPopover',
+    (
+      $http
+      $compile
+      $templateCache
+    ) ->
+      default_template = '''
+        <div class="popover-container">
+          <h3
+            ng-if = "foo.title"
+            class = "popover-title"
+          > {{ foo.title }} </h3>
+          <p>{{ foo.text }}</p>
+        </div>
+      '''
+
+      ###
+      # Directive definition object
+      ###
+
+      restrict: 'A'
+      scope:
+        text: '@shiftPopover'
+        title: '@shiftPopoverTitle'
+        position: '@shiftPopoverPosition'
+        templateUrl: '@shiftPopoverTemplateUrl'
+        trigger: '@shiftPopoverTrigger'
+        attachTo: '@shiftPopoverattachTo'
+
+      link: (scope, element, attrs, controllers, transclude) ->
+        scope.parent_node = element[0]
+
+        do compile = (template=default_template) ->
+          shift_floating = angular.element '<shift-floating />'
+
+          scope.foo = {title: scope.title, text: scope.text}
+          console.log scope.title
```

Comment:
Assuming you were trying to debug why the `ng-if="title"` in the default template wasn't displaying the title when one was provided ;D

I was also stumped